runtime/src/accounts_db.rs

```diff
@@ -701,6 +701,74 @@ impl AccountsDB {
         accounts_index.uncleaned_roots.clear();
     }
 
+    fn inc_store_counts(
+        id: AppendVecId,
+        purges: &HashMap<Pubkey, Vec<(Slot, AccountInfo)>>,
+        store_counts: &mut HashMap<AppendVecId, usize>,
+        already_counted: &mut HashSet<AppendVecId>,
+    ) {
+        if already_counted.contains(&id) {
+            return;
+        }
+        *store_counts.get_mut(&id).unwrap() += 1;
+        let mut keys = HashSet::new();
+        for (key, account_infos) in purges {
+            for (_slot, account_info) in account_infos {
+                if account_info.store_id == id {
+                    keys.insert(key);
```

Comment:
can add `break` here?

Reply:
Don't think so. it needs to find all affected keys by this storage id.

Reply:
ah, you're right, from the inner loop it can break.